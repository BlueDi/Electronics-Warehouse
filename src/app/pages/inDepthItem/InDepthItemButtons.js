import React from 'react';
import { Button } from 'semantic-ui-react';
import '@common/styles/global.css';
import './styles/InDepthItem.scss';

export function InDepthItemButtons(props) {
  const requestButton = (
    <div className="RequestButton" style={{ textAlign: 'right' }}>
      <Button
        onClick={props.handleRequest}
        style={{
          backgroundColor: '#89DF89',
          padding: '10px 15px',
          borderRadius: '10px',
          border: '0px'
        }}
      >
        Request
      </Button>
    </div>
  );

  let editButton = (
    <div className="EditButton" style={{ textAlign: 'left' }}>
      <Button
        onClick={props.handleEdit}
        style={{
          backgroundColor: '#7bbfe8',
          padding: '10px 15px',
          borderRadius: '10px',
          border: '0px'
        }}
      >
        Edit
      </Button>
    </div>
  );

  let cancelButton = (
    <div className="CancelButton" style={{ textAlign: 'left' }}>
      <Button
        onClick={props.handleCancel}
        style={{
          backgroundColor: '#D2E0E8',
          padding: '10px 15px',
          borderRadius: '10px',
          border: '0px'
        }}
      >
        Cancel
      </Button>
    </div>
  );

  let acceptButton = (
    <div className="AcceptButton" style={{ textAlign: 'left' }}>
      <Button
        onClick={props.handleAccept}
        style={{
          backgroundColor: '#52e852',
          padding: '10px 15px',
          borderRadius: '10px',
          border: '0px'
        }}
      >
        Accept
      </Button>
    </div>
  );

  return (
    <div>
      {!props.editing && requestButton}
      {!props.editing && editButton}
      {props.editing && cancelButton}
      {props.editing && acceptButton}
    </div>
  );
}
