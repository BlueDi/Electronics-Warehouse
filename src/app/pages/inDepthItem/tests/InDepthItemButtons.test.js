import React from 'react';
import renderer from 'react-test-renderer';
import InDepthItemButtons from '@pages/inDepthItem/InDepthItemButtons';

describe('<InDepthItemButtons />', () => {
  test('Renders properly', () => {
    const component = renderer.create(<InDepthItemButtons />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders with edit enabled', () => {
    const component = renderer.create(<InDepthItemButtons editing={true} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
