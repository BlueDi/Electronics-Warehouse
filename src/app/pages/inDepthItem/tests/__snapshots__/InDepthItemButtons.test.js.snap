// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InDepthItemButtons /> Renders properly 1`] = `null`;

exports[`<InDepthItemButtons /> Renders with edit enabled 1`] = `null`;
